### PR TITLE
Fixes some flaky tests

### PR DESCRIPTION
**Pull request by pbetkier**
Some tests would randomly fail without reason, this PR should fix this.

### CODE REVIEW

File: consul/consul_test_server.go [hunk 1]
```diff
@@ -6,18 +6,26 @@ import (
 	"time"
 
 	"github.com/hashicorp/consul/testutil"
+	"sync/atomic"
```
Comment (janisz) [inline consul/consul_test_server.go]: Pelase, run [`goimports`](https://godoc.org/golang.org/x/tools/cmd/goimports) on changed files.

**Comment by xor-xor** [comment]
LGTM

File: consul/consul_test_server.go [hunk 2]
```diff
@@ -12,9 +13,43 @@ import (
...
+	defer func() {
+		for _, l := range listener {
+			l.Close()
+		}
+
+	}()
+	for i := 0; i < number; i++ {
+		addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+		if err != nil {
+			panic(err)
```
Comment (xor-xor) [inline consul/consul_test_server.go]: What about returning this error instead of panicking..?

File: consul/consul_test_server.go [hunk 3]
```diff
@@ -12,9 +13,43 @@ import (
 var dcOffset uint32 = 1
 var failingClientOffset uint32 = 1
 
+// Ask the kernel for a free open ports that are ready to use
```
Comment (xor-xor) [inline consul/consul_test_server.go]: s/for a free/for free/

Comment (janisz) [inline consul/consul_test_server.go]: Thanks, fixed

File: consul/consul_test_server.go (hunk 2 above)
Comment (janisz) [inline consul/consul_test_server.go]: Good point. Fixed it.

**Comment by xor-xor** [comment]
:ok:

**Comment by janisz** [comment]
LGTM :+1:

**Comment by dankraw** [comment]
👏 👏 👏

**Pull request merged**